xwayland/xwm.c

```diff
@@ -243,6 +245,24 @@ static void xwm_set_net_client_list(struct wlr_xwm *xwm) {
 			XCB_ATOM_WINDOW, 32, mapped_surfaces, windows);
 }
 
+static void xwm_set_net_client_list_stacking(struct wlr_xwm *xwm) {
+	size_t num_surfaces = wl_list_length(&xwm->surfaces_in_stack_order);
+	xcb_window_t windows[num_surfaces + 1];
```

Comment:
A client can blow up our stack by opening many windows.

Reply:
Added a commit fixing this and the other instance for `_NET_CLIENT_LIST`.